### PR TITLE
[CT-1953]: convert redshift_test to functional test

### PR DESCRIPTION
**Pull request by nssalian**
resolves #

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-redshift/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-redshift/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: tests/functional/adapter/redshift_test/test_late_binding_view.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+from dbt.tests.util import run_dbt, run_sql_with_adapter
+
+_MODEL_SQL = """{{
+  config(
+    materialized='view',
+    bind=False
+  )
+}}
+select * from {{ ref('seed') }}
+
```
Comment (VersusFacit) [inline tests/functional/adapter/redshift_test/test_late_binding_view.py:12]: nit: whitespace

File: tests/functional/adapter/redshift_test/test_late_binding_view.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+"""
+
+_SEED_CSV = """
+id,first_name,email,ip_address,updated_at
+1,Larry,lking0@miitbeian.gov.cn,69.135.206.194,2008-09-12 19:08:31
+""".lstrip()
```
Comment (VersusFacit) [inline tests/functional/adapter/redshift_test/test_late_binding_view.py:18]: great id I'm going to steal in the future

**Review by VersusFacit — APPROVED** [review]
LGTM, just left some intended to be instructive comments for the test suite/formatting stuff

File: tests/functional/adapter/redshift_test/test_late_binding_view.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            'seeds': {
+                'quote_columns': False,
+            }
+        }
+
+    def test__redshift_late_binding_view_query(self, project):
```
Comment (VersusFacit) [inline tests/functional/adapter/redshift_test/test_late_binding_view.py:43]: nit: we don't need to keep `redshift` in test names anymore

File: tests/functional/adapter/redshift_test/test_late_binding_view.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        seed_run_result = run_dbt(['seed'])
+        assert len(seed_run_result) == 1
+        run_result = run_dbt()
+        assert len(run_result) == 1
+        # drop the table. Use 'cascade' here so that if late-binding views
+        # didn't work as advertised, the following dbt run will fail.
+        drop_query = """drop table if exists {}.seed cascade""".format(project.test_schema)
+        run_sql_with_adapter(project.adapter, drop_query)
```
Comment (VersusFacit) [inline tests/functional/adapter/redshift_test/test_late_binding_view.py:51]: nit: alternate option `project.run_sql` if I'm not mistaken

**Review by colin-rogers-dbt — APPROVED** [review]

**Comment by nssalian** [comment]
Thanks for the review. Merging as soon as tests pass on the latest commit.

**Pull request merged**